File: dsa/easy/First Non Repeating Character/first_non_repeating_character.cpp
#include <string>
#include <unordered_map>

int firstNonRepeatingCharacter(std::string s) {
  std::unordered_map<char, int> charCount;
  for (int i = 0; i < s.length(); i++) {
    charCount[s[i]]++;
  }
  for (int i = 0; i < s.length(); i++) {
    if (charCount[s[i]] == 1) {
      return i;
    }
  }
  return -1;
}

File: dsa/easy/First Non Repeating Character/first_non_repeating_character.go
package main

func FirstNonRepeatingCharacter(str string) int {
  if len(str) == 0 {
    return -1
  }
  m := make(map[rune]int)
  for _, c := range str {
    m[c]++
  }
  for i, c := range str {
    if m[c] == 1 {
      return i
    }
  }
  return -1
}

